src/BreakfastSerializer/Serializer.php

```diff
@@ -0,0 +1,224 @@
+<?php
+
+namespace BDBStudios\BreakfastSerializer;
+
+/**
+ * Class Serializer
+ * @package BDBStudios
+ */
+class Serializer implements Serializable
+{
+
+    /**
+     * @var int
+     */
+    protected $format;
+
+    /**
+     * @param int $dataFormat
+     */
+    protected function __construct($dataFormat = Serializer::FORMAT_XML)
+    {
+        $this->format = $dataFormat;
+    }
+
+    /**
+     * @return int
+     */
+    public function getFormat()
+    {
+        return $this->format;
+    }
+
+    /**
+     * @param int $dataFormat
+     * @return Serializer
+     */
+    public static function getSerializer($dataFormat = Serializable::FORMAT_JSON)
+    {
+        $instance = new self($dataFormat = Serializable::FORMAT_JSON);
+
+        return $instance;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public static function isIterable($object)
+    {
+        return (is_array($object) || $object instanceof \Traversable);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function deserialize($data, $dataFormat = Serializable::FORMAT_JSON)
+    {
+        if (Serializable::FORMAT_JSON !== $dataFormat) {
+            throw new \LogicException('Currently only JSON is supported');
+        }
+
+        switch ($dataFormat) {
+            case Serializable::FORMAT_JSON:
+                $data = json_decode($data, true);
+                return $this->arrayToObject(
+                    $data
+                );
+        }
+    }
+
+    /**
+     * @param object $data
+     * @param int $dataFormat
+     * @return mixed
+     */
+    public function serialize($data, $dataFormat = Serializable::FORMAT_JSON)
+    {
+        if (Serializable::FORMAT_JSON !== $dataFormat) {
+            throw new \LogicException('Currently only JSON is supported');
+        }
+
+        switch ($dataFormat) {
+            case Serializable::FORMAT_JSON:
+                return json_encode(
+                    $this->objectToArray(
+                        $data
+                    )
+                );
+        }
+    }
+
+    /**
+     * @param array $data
+     * @return mixed
+     * @throws \Exception
+     */
+    protected function arrayToObject(array $data)
+    {
+        $object     = new $data['className']();
+        $reflection = new \ReflectionClass($data['className']);
+        $breadth    = array();
+
+        $object = $this->extractAndSetSingleDepthProperties($data, $breadth, $reflection, $object);
+        $object = $this->extractAndSetMultipleDepthProperties($breadth, $reflection, $object);
+
+        return $object;
+    }
+
+    /**
+     * @param array $data
+     * @param array $breadth
+     * @param \ReflectionClass $reflection
+     * @param mixed $object
+     * @return mixed
+     * @throws \Exception
+     */
+    protected function extractAndSetSingleDepthProperties(
+        array $data,
+        array& $breadth,
+        \ReflectionClass& $reflection,
+        $object
+    )
```

Comment:
it will be once I start implementing the other serializers, this will be in my base class